src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs

```diff
@@ -1,408 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System.Collections.Immutable;
-using System.Diagnostics;
-using System.Threading;
-using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Host;
-using Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics;
-using Microsoft.CodeAnalysis.PooledObjects;
-using Microsoft.CodeAnalysis.Shared.Extensions;
-using Microsoft.CodeAnalysis.SolutionCrawler;
-using Microsoft.CodeAnalysis.Workspaces.Diagnostics;
-using Roslyn.Utilities;
-
-namespace Microsoft.CodeAnalysis.Diagnostics;
-
-internal partial class DiagnosticAnalyzerService
-{
-    private partial class DiagnosticIncrementalAnalyzer
-    {
-        /// <summary>
-        /// State for diagnostics that belong to a project at given time.
-        /// </summary>
-        private sealed class ProjectState
-        {
-            private const string SyntaxStateName = nameof(SyntaxStateName);
-            private const string SemanticStateName = nameof(SemanticStateName);
-            private const string NonLocalStateName = nameof(NonLocalStateName);
-
-            // project id of this state
-            private readonly StateSet _owner;
-
-            // last aggregated analysis result for this project saved
-            private DiagnosticAnalysisResult _lastResult;
-
-            public ProjectState(StateSet owner, ProjectId projectId)
-            {
-                _owner = owner;
-                _lastResult = DiagnosticAnalysisResult.CreateInitialResult(projectId);
```

Comment:
_lastResult woudl stay this value forever (until ForceAnalyzeProjectAsync was called).  So keep that in mind for the majorit of code that exists in this type as the review continues below.